packages/backend-function/src/resolve_secret_banner.ts

```diff
@@ -0,0 +1,31 @@
+import { SSM } from '@aws-sdk/client-ssm';
```

Comment:
How does bundler handle this import?
The reason I'm asking is that for other files that are bundled I'm guessing bunder generates some form of import/require in right place, but I'm not sure how this works for banner (which they advertise as a place to inject comments), so I wonder if theres a chance that in some more complex lambdas this might become a problem.
Might be helpful if you could share how sample bundled code looks like in a comment.

Reply:
Bundling format is `ESM` to allow the top level await to work. The code with an example handler in lambda console looks like this:
```
import { SSM } from '@aws-sdk/client-ssm';/** * The body of this function will be used to resolve secrets for Lambda functions */export const resolveSecretBanner = async (client) => {    if (!client) {        client = new SSM();    }    const envPathObj = JSON.parse(process.env.AMPLIFY_SECRET_PATHS ?? '{}');    const paths = Object.values(envPathObj);    if (paths.length > 0) {        const response = await client.getParameters({            Names: paths,            WithDecryption: true,        });        if (response.Parameters && response.Parameters?.length > 0) {            for (const parameter of response.Parameters) {                for (const [env, path] of Object.entries(envPathObj)) {                    if (parameter.Name === path) {                        process.env[env] = parameter.Value;                    }                }            }        }    }};await resolveSecretBanner();//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoicmVzb2x2ZV9zZWNyZXRfYmFubmVyLmpzIiwic291cmNlUm9vdCI6IiIsInNvdXJjZXMiOlsiLi4vc3JjL3Jlc29sdmVfc2VjcmV0X2Jhbm5lci50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxPQUFPLEVBQUUsR0FBRyxFQUFFLE1BQU0scUJBQXFCLENBQUM7QUFFMUM7O0dBRUc7QUFDSCxNQUFNLENBQUMsTUFBTSxtQkFBbUIsR0FBRyxLQUFLLEVBQUUsTUFBWSxFQUFFLEVBQUU7SUFDeEQsSUFBSSxDQUFDLE1BQU0sRUFBRTtRQUNYLE1BQU0sR0FBRyxJQUFJLEdBQUcsRUFBRSxDQUFDO0tBQ3BCO0lBQ0QsTUFBTSxVQUFVLEdBQTJCLElBQUksQ0FBQyxLQUFLLENBQUMsT0FBTyxDQUFDLEdBQUcsQ0FBQyxvQkFBb0IsSUFBSSxJQUFJLENBQUMsQ0FBQztJQUNoRyxNQUFNLEtBQUssR0FBRyxNQUFNLENBQUMsTUFBTSxDQUFDLFVBQVUsQ0FBQyxDQUFDO0lBQ3hDLElBQUksS0FBSyxDQUFDLE1BQU0sR0FBRyxDQUFDLEVBQUU7UUFDcEIsTUFBTSxRQUFRLEdBQUcsTUFBTSxNQUFNLENBQUMsYUFBYSxDQUFDO1lBQzFDLEtBQUssRUFBRSxLQUFLO1lBQ1osY0FBYyxFQUFFLElBQUk7U0FDckIsQ0FBQyxDQUFDO1FBQ0gsSUFBSSxRQUFRLENBQUMsVUFBVSxJQUFJLFFBQVEsQ0FBQyxVQUFVLEVBQUUsTUFBTSxHQUFHLENBQUMsRUFBRTtZQUMxRCxLQUFLLE1BQU0sU0FBUyxJQUFJLFFBQVEsQ0FBQyxVQUFVLEVBQUU7Z0JBQzNDLEtBQUssTUFBTSxDQUFDLEdBQUcsRUFBQyxJQUFJLENBQUMsSUFBSSxNQUFNLENBQUMsT0FBTyxDQUFDLFVBQVUsQ0FBQyxFQUFFO29CQUNuRCxJQUFJLFNBQVMsQ0FBQyxJQUFJLEtBQUssSUFBSSxFQUFFO3dCQUMzQixPQUFPLENBQUMsR0FBRyxDQUFDLEdBQUcsQ0FBQyxHQUFHLFNBQVMsQ0FBQyxLQUFLLENBQUM7cUJBQ3BDO2lCQUNGO2FBQ0Y7U0FDRjtLQUNGO0FBQ0gsQ0FBQyxDQUFDO0FBRUYsTUFBTSxtQkFBbUIsRUFBRSxDQUFDIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IHsgU1NNIH0gZnJvbSAnQGF3cy1zZGsvY2xpZW50LXNzbSc7XG5cbi8qKlxuICogVGhlIGJvZHkgb2YgdGhpcyBmdW5jdGlvbiB3aWxsIGJlIHVzZWQgdG8gcmVzb2x2ZSBzZWNyZXRzIGZvciBMYW1iZGEgZnVuY3Rpb25zXG4gKi9cbmV4cG9ydCBjb25zdCByZXNvbHZlU2VjcmV0QmFubmVyID0gYXN5bmMgKGNsaWVudD86IFNTTSkgPT4ge1xuICBpZiAoIWNsaWVudCkge1xuICAgIGNsaWVudCA9IG5ldyBTU00oKTtcbiAgfVxuICBjb25zdCBlbnZQYXRoT2JqOiBSZWNvcmQ8c3RyaW5nLCBzdHJpbmc+ID0gSlNPTi5wYXJzZShwcm9jZXNzLmVudi5BTVBMSUZZX1NFQ1JFVF9QQVRIUyA/PyAne30nKTtcbiAgY29uc3QgcGF0aHMgPSBPYmplY3QudmFsdWVzKGVudlBhdGhPYmopO1xuICBpZiAocGF0aHMubGVuZ3RoID4gMCkge1xuICAgIGNvbnN0IHJlc3BvbnNlID0gYXdhaXQgY2xpZW50LmdldFBhcmFtZXRlcnMoe1xuICAgICAgTmFtZXM6IHBhdGhzLFxuICAgICAgV2l0aERlY3J5cHRpb246IHRydWUsXG4gICAgfSk7XG4gICAgaWYgKHJlc3BvbnNlLlBhcmFtZXRlcnMgJiYgcmVzcG9uc2UuUGFyYW1ldGVycz8ubGVuZ3RoID4gMCkge1xuICAgICAgZm9yIChjb25zdCBwYXJhbWV0ZXIgb2YgcmVzcG9uc2UuUGFyYW1ldGVycykge1xuICAgICAgICBmb3IgKGNvbnN0IFtlbnYscGF0aF0gb2YgT2JqZWN0LmVudHJpZXMoZW52UGF0aE9iaikpIHtcbiAgICAgICAgICBpZiAocGFyYW1ldGVyLk5hbWUgPT09IHBhdGgpIHtcbiAgICAgICAgICAgIHByb2Nlc3MuZW52W2Vudl0gPSBwYXJhbWV0ZXIuVmFsdWU7XG4gICAgICAgICAgfVxuICAgICAgICB9XG4gICAgICB9XG4gICAgfVxuICB9XG59O1xuXG5hd2FpdCByZXNvbHZlU2VjcmV0QmFubmVyKCk7XG4iXX0=

// test-projects/testapp/amplify/function/handler.ts
var handler = async (event) => {
  console.log(`TEST_SECRET is ${process.env.TEST_SECRET}`);
  console.log(`TEST_SECRET2 is ${process.env.TEST_SECRET2}`);
};
export {
  handler
};
```

Reply:
If we're going to minify like this, we should remove all whitespace. We should also remove the source map

Reply:
is this working fine if `test-projects/testapp/amplify/function/handler.ts` has it's own imports as well?

Reply:
Not sure of a good way to remove all the meaningless whitespace (we would need whitespace between `await` and `internalAmplifyFunctionBannerResolveSecrets();` for example)

Reply:
> is this working fine if `test-projects/testapp/amplify/function/handler.ts` has its own imports as well?

Yes, all the contents of `handler.ts` will be under the comment `// test-projects/testapp/amplify/function/handler.ts` and still work as expected from what I've tested.

Reply:
There are some interesting gotchas for this file. For example, an inline comment `// some comment` would break everything after that after it gets collapsed into a single line. I think imports of 3p modules or other files in resolve_secret_banner would also break as we are not following those imports and bundling them as well.

The "real" fix for all of this would be to run resolve_secret_banner through esbuild, then add that bundled output as the banner for esbuild of the customer lambda. But I'm okay if we just want to capture this thread as a GH issue and revisit if/when we need to do more complicated stuff here.

Reply:
It sounds like there was some windows / unix issue related with keeping the newlines, but if we kept newlines in, that would remove the inline comment gotcha and make the output more readable.